akka-actor/src/main/scala/akka/dispatch/Future.scala

```diff
@@ -74,6 +69,20 @@ object ExecutionContexts {
    */
   def global(): ExecutionContextExecutor = ExecutionContext.global
 
+  /**
+   * WARNING: Not A General Purpose ExecutionContext!
+   *
+   * This is an execution context which runs everything on the calling thread.
+   * It is very useful for actions which are known to be non-blocking and
+   * non-throwing in order to save a round-trip to the thread pool.
+   *
+   * Alias for [[sameThreadExecutionContext]], which is deprecated in 2.6, providing cross version non-deprecated access.
+   *
+   * INTERNAL API
+   */
+  @InternalApi
```

Comment:
Should we make it `@InternalStableAPI`?

Reply:
So far I think we've only used that for things used by cinnamon but it makes sense for APIs used across the Akka projects as well.